library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -21,18 +21,47 @@ PlatformBridgeFilterConfig::PlatformBridgeFilterConfig(
           Api::External::retrieveApi(proto_config.platform_filter_name()))) {}
 
 PlatformBridgeFilter::PlatformBridgeFilter(PlatformBridgeFilterConfigSharedPtr config)
-    : platform_filter_(config->platform_filter()) {}
+    : platform_filter_(*config->platform_filter()) {
+  // The initialization above sets platform_filter_ to a copy of the struct stored on the config.
+  // In the typical case, this will represent a filter implementation that needs to be intantiated.
+  // static_context will contain the necessary platform-specific mechanism to produce a filter
+  // instance. instance_context will initially be null, but after initialization, set to the
+  // context needed for actual filter invocations.
+
+  // We treat a null initializer as a static implementation.
+  if (platform_filter_.init_filter == nullptr) {
+    // Make static context available to invocations, in lieu of instatiated context.
+    platform_filter_.instance_context = platform_filter_.static_context;
+    return;
+  }
+
+  // Set the instance_context to the result of the initialization call. Cleanup will ultimately
+  // occur during in the onDestroy() invocation below.
+  platform_filter_.instance_context = platform_filter_.init_filter(platform_filter_.static_context);
+
+  ASSERT(platform_filter_.instance_context, "init_filter unsuccessful");
```

Comment:
can we interpolate the filter name here? Makes for easier crash debugging

Reply:
Sure.